### PR TITLE
Add recipe for Apache Storm 0.10.1

### PR DESCRIPTION
**Pull request by tdhopper**

### CODE REVIEW

**Comment by tdhopper** [comment]
Hmm. Can anyone help here? It seems to work for me locally. Storm ships precompiled binaries, so I'm just coping them into the environment path.

**Comment by jakirkham** [comment]
> Storm ships precompiled binaries, so I'm just coping them into the environment path.

Why are we using precompiled binaries?

**Comment by tdhopper** [comment]
@jakirkham: Err? Less work for me.

**Comment by jakirkham** [comment]
So, we prefer to build things from source here. Is that not possible for some technical reason?

**Comment by tdhopper** [comment]
PEBCAK error, I guess.

**Comment by tdhopper** [comment]
I know nothing about building JVM projects. If someone wants to jump in and help, I'm happy to learn. Otherwise, I'll just close this.

**Comment by jakirkham** [comment]
Here are the [instructions]( https://github.com/apache/storm/blob/master/DEVELOPER.md#building ). Looks pretty simple. Do you want to give it a try first?

**Comment by danielfrg** [comment]
Just FYI, I have used spark precompiled binaries to create a conda pkgs for spark with success.

For these apache (specially hadoop related) projects I believe this is the best way to create conda pkgs since the build process its not only complex because it involves java but this projects are so big (in dependencies and other things) that not a lot of people know how to build them so trusting the maintainers its the best way to get this packages installed.

As an example for spark there is different versions that match different version of hadoop and the maintainers provide tested pkgs for those.

File: recipes/apache-storm/build.sh [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+
+ rsync -a $SRC_DIR/* $PREFIX
```
Comment (ccordoba12) [inline recipes/apache-storm/build.sh]: Is not possible to use `cp -R` here?

File: recipes/apache-storm/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  sha256: f9b76eb8cb59b530888aea4b957292c510bae2d9ada48c705d6f567d4dcc74bc
+
+build:
+  number: 0
+  skip: True  # [win]
+
+requirements:
+  build:
+    - 'curl'
+  run: []
```
Comment (ccordoba12) [inline recipes/apache-storm/meta.yaml]: Is this (empty) section necessary?

**Comment by ccordoba12** [comment]
I'm +1 to use precompiled binaries for now. Maybe later the feedstock maintainers would want to compile the package directly, but that's up to them :-)

@tdhopper, the failure on Linux (CircleCI) could be caused because the docker image doesn't define the USER environment variable. So if `storm` needs to read it to define a configuration value, that will fail :-)

The solution is to run the test as

    - USER=test storm version  # [linux]

But I don't know if that will help :-)

File: recipes/apache-storm/build.sh (hunk 1 above)
Comment (ccordoba12) [inline recipes/apache-storm/build.sh]: I mean, what's the advantage of using `rsync`

**Pull request closed without merging**